File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
private:
    void rView(TreeNode* node, vector<int> &ans, int level){

        if(node==nullptr) return;
        if(level == ans.size()) ans.push_back(node->val);
        rView(node->right,ans,level+1);
        rView(node->left,ans,level+1);
    }

public:
    vector<int> rightSideView(TreeNode* root) {

        vector<int> ans;
        if(root==nullptr) return ans;
        rView(root,ans,0);
        return ans;


        
    }
};

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def rightSideView(self, root):
        """
        :type root: Optional[TreeNode]
        :rtype: List[int]
        """

        res=[]
        self.recursion(root,0,res)
        return res

    def recursion(self,node,level,res):
        
        if node is None:
            return

        if level==len(res):
            res.append(node.val)

        self.recursion(node.right,level+1,res)
        self.recursion(node.left,level+1,res)
